compiler-core/src/parse.rs

```diff
@@ -2128,10 +2128,30 @@ where
             let constructors = Parser::series_of(
                 self,
                 &|p| {
+                    // The only attribute supported on constructors is @deprecated
+                    let mut attributes = Attributes::default();
+                    let attr_loc = Parser::parse_attributes(p, &mut attributes)?;
+
+                    // Expecting all but the deprecated atterbutes to be default
+                    if attr_loc.is_some() // only if attributes are present
+                        && (attributes.external_erlang.is_some()
+                            || attributes.external_javascript.is_some()
+                            || attributes.target.is_some()
+                            || attributes.internal != InternalAttribute::Missing)
+                    {
+                        let attr_span = attr_loc
+                            .expect("RecordConstructor musn't be None already checked if None");
```

Comment:
Always pattern match rather than checking a bool expression and then checking it a second time with `expect`